p2panda-rs/src/schema/schema_id.rs

```diff
@@ -87,20 +85,21 @@ impl<'de> Visitor<'de> for SchemaIdVisitor {
     where
         S: SeqAccess<'de>,
     {
-        let mut hashes: Vec<Hash> = Vec::new();
-
-        while let Some(hash) = seq.next_element::<Hash>()? {
-            if hash.validate().is_err() {
-                return Err(serde::de::Error::custom(format!(
-                    "Invalid hash {:?}",
-                    hash.as_str()
-                )));
-            }
-
-            hashes.push(hash);
+        let mut op_ids: Vec<OperationId> = Vec::new();
+
+        while let Some(seq_value) = seq.next_element::<String>()? {
+            match seq_value.parse::<OperationId>() {
+                Ok(operation_id) => op_ids.push(operation_id),
+                Err(hash_err) => {
+                    return Err(serde::de::Error::custom(format!(
+                        "Error parsing application schema id: {}",
+                        hash_err
+                    )))
+                }
+            };
```

Comment:
Ah interesting, I'm not sure if I understand it correctly now, but does your change still check for the validity of a hash? Deserializing something is simply just moving a string into `Hash` without checking if it fullfils the format of YASMF. This is where `validate` comes in to prevent a random string to move in. Do you account somewhere else for this now? Or maybe I'm getting it wrong ..

Reply:
Yes, previously the `Hash` constructor would fail for invalid values because it invokes validation. The previous code was invoking `validate()` again separately afterwards but those calls would always go through because the validity had already been checked in the constructor.

Reply:
`while let Some(hash) = seq.next_element::<Hash>()? {` is not calling a constructor though if I understand it correctly.

Reply:
Created an issue for the general question: https://github.com/p2panda/p2panda/issues/276